Fix issue #41 Different ply returned by same position

gbtami opened the pull request:
Use nth ply move from .pgn in nth ply position in move related rules.

gbtami committed:
Fix issue #41 Different ply returned by same position
Use nth ply move from .pgn in nth ply position in move related rules.

mcostalba commented:
This is done on purpose, if you see:

https://github.com/mcostalba/scoutfish/blob/master/src/scout.cpp#L368

Here the move is coded as (piece, to_square), so that _Nf3_ picks the first knight move in _f3_ and does not matter where the knight comes from: algebraic notation forces to explicit the from square but IMO is too restrictive.

Eventually I will add a _disambiguation_ flag, to correctly handle cases where disambiguation is given but without being too restrictive on all the others.

gbtami committed:
Merge branch 'master' into master

gbtami committed:
In starting position (ply=0) the pgn move (ply=0) is always MOVE_NONE

gbtami committed:
Fix makefile x86-32 target

gbtami committed:
Revert prev commit

gbtami committed:
Add some build stuff based on nfiekas/Stockfish

gbtami committed:
Don't use docker on build

gbtami committed:
Add g++-multilib to travis install

gbtami committed:
Add virtualenv to travis install

gbtami committed:
Use pexpect from virtualenv

gbtami committed:
Drop Windows build until time being

gbtami committed:
Add deploy to travis

gbtami committed:
Remove branch whitelisting to enable deploy

gbtami committed:
Drop Windows build until time being

gbtami committed:
Add appveyor.yml

gbtami committed:
Fix appveyor test.py run

gbtami committed:
Build x86_64-modern also

gbtami committed:
Changed repo ownership from gbtami to pychess

gbtami committed:
Add travis build status badge

gbtami committed:
Add failing testcase to https://github.com/pychess/pychess/issues/1550

alip committed:
search: add braces around do_move

clang-10.0.0 warns about this misleading indentation as follows:

> scout.cpp: In function 'void Scout::search(Thread*)':
> scout.cpp:302:11: warning: this 'if' clause does not guard... [-Wmisleading-indentation]
>   302 |           if (move)
>       |           ^~
> scout.cpp:304:15: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the '
> if'
>   304 |               pos.do_move(move, *st++, pos.gives_check(move));
>       |               ^~~

This looks like a bug to me as the condition `move==MOVE_NONE` will
cause a NULL point dereference in `do_move`.

Signed-off-by: Ali Polatel <alip@exherbo.org>

gbtami committed:
Merge pull request #6 from alip/search-braces-around-do-move

search: add braces around do_move
